### PR TITLE
Julia language parser.

### LINKED ISSUE #8: julia language support
Are there any plans to support [julia](https://github.com/tree-sitter/tree-sitter-julia)? I've coded this  function

``` emacs-lisp
(defun tree-sitter-fold-range-julia (node)
  (let* ((a (tsc-node-end-position (tsc-get-nth-named-child node 0)))
         (b (- (tsc-node-end-position node) 3))) ;; offset by three to include `end'
    (cons a b)))
```
for the [tree-sitter-fold](https://github.com/junyi-hou/tree-sitter-fold) package to fold the `function_definition`, `macro_definition`, `struct_definition` and `module_definition` nodes; it works, but it sometimes fails to load (I don't know why).

Also, the code here looks much more comprehensive (and standardized); I also don't know enough elisp to understand it or contribute with my own julia parser. Since julia uses the `function ... end` syntax, I feel it shouldn't be too different from ruby (maybe?)

Thanks.

### PR DESCRIPTION
**Pull request by SrAceves**
Closes #8.

### CODE REVIEW

File: ts-fold-parsers.el [hunk 1]
```diff
@@ -242,5 +242,20 @@
...
+    (function_definition . ts-fold-range-julia)
+    (if_statement        . (ts-fold-range-seq 1 -2))
+    (let_statement       . (ts-fold-range-seq 2 -2))
+    (macro_definition    . ts-fold-range-julia)
+    (module_definition   . ts-fold-range-julia)
+    (quote_statement     . ts-fold-range-julia)
+    (struct_definition   . ts-fold-range-julia)
+    (triple_string       . (ts-fold-range-seq 2 -2))
+    (try_statement       . (ts-fold-range-seq 2 -2))
+    (while_statement     . ts-fold-range-julia)))
```
Comment (jcs090218) [inline ts-fold-parsers.el:258]: Can you place this function in alphabetic order? Thanks! ❤️

File: ts-fold-summary.el [hunk 2]
```diff
@@ -195,37 +199,38 @@ type of content by checking the word boundary's existence."
...
+  (ruby-mode         . ts-fold-summary-ruby-doc)
+  (rust-mode         . ts-fold-summary-rust-doc)
+  (scala-mode        . ts-fold-summary-javadoc)
+  (sh-mode           . ts-fold-summary-javadoc)
+  (swift-mode        . ts-fold-summary-c)
+  (typescript-mode   . ts-fold-summary-javadoc)
+  (nxml-mode         . ts-fold-summary-xml))
+"Alist mapping `major-mode' to doc parser function."
+:type 'hook
+:group 'ts-fold)
```
Comment (jcs090218) [inline ts-fold-summary.el:233]: Can you fix the indentation here? Thanks! 🚀

**Review by jcs090218 — COMMENTED** [review]
I have planned to work on this but you have made it first! ❤️ That's awesome! Thanks!

Some nits:

1. Can you update the README.md? Add `Julia` to the supported languages list should be enough.
2. Update the CHANGELOG.md file.
3. Address all code review issues (see the review).

File: ts-fold-parsers.el (hunk 1 above)
Comment (SrAceves) [reply ts-fold-parsers.el:258]: looks in alphabetic order to me (I actually ran `sort-lines` on the region); or you mean the associated functions?

Comment (jcs090218) [reply ts-fold-parsers.el:258]: Oh, yes. The associated functions. The node doesn't have to be in alphabetic order.

File: ts-fold-parsers.el [hunk 3]
```diff
@@ -242,5 +242,20 @@
   "Rule set for TypeScript."
   (append (ts-fold-parsers-javascript)))
 
+(defun ts-fold-parsers-julia ()
+  "Rule set for Julia."
+  '((block_comment       . (ts-fold-range-seq 1 -1))
+    (for_statement       . (ts-fold-range-seq 2 -2))
+    (function_definition . ts-fold-range-julia)
```
Comment (jcs090218) [inline ts-fold-parsers.el:249]: You might need to forward declare the function `ts-fold-range-julia` at the top of the file. Like these,

https://github.com/emacs-tree-sitter/ts-fold/blob/28409a0ceede0751ed9d520c6a19d1f5f1211502/ts-fold-parsers.el#L38-L53

**Comment by jcs090218** [comment]
Let's try to avoid this `checkdoc` warning as well, see https://github.com/emacs-tree-sitter/ts-fold/runs/8174658230?check_suite_focus=true#step:6:170.

```
`ts-fold.el` with checkdoc (0.6.2)
ts-fold.el:602: There should be two spaces after a period
```

File: ts-fold-parsers.el (hunk 1 above)
Comment (jcs090218) [reply ts-fold-parsers.el:259]: I think I misunderstood associated functions, you should place the whole function `ts-fold-parsers-julia` under `ts-fold-parsers-json`, and not the very end of the file (not under `ts-fold-parsers-typescript`).

**Review by jcs090218 — APPROVED** [review]
LGTM, just place the whole function `ts-fold-parsers-julia` under `ts-fold-parsers-json` (alphabetic order), then it should be good to go?

Let me know if this PR is ready to merge!

**Comment by SrAceves** [comment]
Everything seems right now!

**Comment by jcs090218** [comment]
Thanks for the changes! ❤️ Merge this now!

**Pull request merged**